grudge/dof_desc.py

```diff
@@ -96,29 +100,39 @@ def __repr__(self):
         return "<{}({})>".format(type(self).__name__, repr(self.tag))
 
 
-class QTAG_NONE:  # noqa: N801
-    """A quadrature tag indicating the use of a
+class DISCR_TAG_BASE:  # noqa: N801
+    """A discretization tag indicating the use of a
     basic discretization grid. This tag is used
-    to distinguish the base discretization (`QTAG_NONE`)
-    from quadrature (e.g. overintegration) grids.
+    to distinguish the base discretization from quadrature
+    (e.g. overintegration) or modal (:class:`DISCR_TAG_MODAL`)
+    discretizations.
     """
 
 
-class QTAG_MODAL:  # noqa: N801
-    """A quadrature tag indicating the use of a
+class DISCR_TAG_QUAD:  # noqa: N801
+    """A discretization tag indicating the use of a
+    quadrature discretization grid. This tag is used
+    to distinguish the quadrature discretization
+    (e.g. overintegration) from modal (:class:`DISCR_TAG_MODAL`)
+    or base (:class:`DISCR_TAG_BASE`) discretizations.
+    """
```

Comment:
Do you want to include a blurb about subclassing to accomplish more than one quadrature tag?

Reply:
Ah I just realized something. So all these tags (`DISCR_TAG_`, `DTAG_`, etc) are never used as instantiated objects. The only exception is `DTAG_BOUNDARY` I think. Do you think `DISCR_TAG_QUAD` should follow a similar behavior? That way we can check `isinstance(tag, DISCR_TAG_QUAD)` if we ever want to check if a quadrature tag is in use (and similarly in other places).

What do you think @inducer ?

Reply:
You could use [issubclass](https://docs.python.org/3/library/functions.html#issubclass) to just check classes, not instances?

Reply:
I think instances for `DTAG_QUAD` make sense. It doesn't preclude subclassing either.

Reply:
How's this: https://github.com/inducer/grudge/pull/83/commits/4adf94e01bc63ce18d2891723829092b5e40f7b8?

Reply:
Relevant commit: https://github.com/inducer/grudge/pull/83/commits/c062ec2fe80eb4953e3113abea17a341fe1a25f6